vaas/vaas/vcl/renderer.py

```diff
@@ -167,10 +167,24 @@ def __init__(self, varnish, input_data):
             'probe': self.prepare_probe(vcl_directors),
             'active_director': active_directors,
             'routes': self.prepare_route(self.varnish, cluster_directors),
+            'redirects': self.prepare_redirect(),
             'cluster_directors': cluster_directors,
             'mesh_routing': varnish.cluster.service_mesh_routing
         }
 
+    @collect_processing
+    def prepare_redirect(self):
+        redirects = {}
+        for redirect in self.input.redirects:
+            if redirect.src_domain in self.varnish.cluster.domainmapping_set.all():
```

Comment:
are these domainmappings prefetched or are you gonna querying them for each redirect separately?

Reply:
The clusters are prefetched when Varnish servers are getting form db. I don't know how we can prefetch there domainmapping. Varnish don't have relations with domains.

Reply:
The quick-win solution is introducing a local variable in the top of function:

```
mappings = self.varnish.cluster.domainmapping_set.all()
```

and use that variable instead calling domainmapping_set.all()